### PR TITLE
[Merged by Bors] - feat: add Tendsto.fst, Tendsto.snd, Tendsto.apply and nhds versions

### PR DESCRIPTION
**Pull request by ADedecker**
- Add `Tendsto.fst`, `Tendsto.snd` and `Tendsto.apply` (these are about product of filters)
- Move the current `Tendsto.apply` to `Tendsto.apply_nhds`, and add `Tendsto.fst_nhds` and `Tendsto.snd_nhds` (these are about neighborhoods in a product space)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Filter/Prod.lean [hunk 1]
```diff
@@ -139,6 +139,20 @@ theorem tendsto_snd : Tendsto Prod.snd (f ×ˢ g) g :=
...
+  tendsto_fst.comp H
+
+/-- If a function tends to a product `g ×ˢ h` of filters, then its second component tends to
+`h`. See also `Filter.Tendsto.snd_nhds` for the special case of converging to a point in a
+product of two topological spaces. -/
+theorem Tendsto.snd {h : Filter γ} {m : α → β × γ} (H : Tendsto m f (g ×ˢ h)) :
+    Tendsto (fun a ↦ (m a).2) f h :=
+  tendsto_snd.comp H
+
 theorem Tendsto.prod_mk {h : Filter γ} {m₁ : α → β} {m₂ : α → γ}
```
Comment (eric-wieser) [inline Mathlib/Order/Filter/Prod.lean:156]: Do we have the nhds version of this?

**Review by eric-wieser — APPROVED** [review]
bors d+

File: Mathlib/Order/Filter/Prod.lean (hunk 1 above)
Comment (ADedecker) [reply Mathlib/Order/Filter/Prod.lean:156]: [Filter.Tendsto.prod_mk_nhds](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Constructions.html#Filter.Tendsto.prod_mk_nhds) ?

**Pull request closed without merging**